### ISSUE TITLE
Feature proposal: tags for organizing conversations

### ISSUE THREAD

**Issue by aramburu** [TokTok/c-toxcore#882]
Hello everyone:

I would like to have some feedback about this idea, and whether it is suitable for Tox.

Usually, groups in messaging apps become flooded of messages that we are not necessarily interested in. My idea is the following: creating the possibility of adding tags to any message in a conversation. 
These system would include the following features:

•	Tags may be added to a message not just when it is sent, but anytime afterwards.
•	Tags may be added to a message not just by the sender, buy by anyone in the group.
•	Tags have a weight or rating, in such a way that different users may confirm that the tag is correctly associated to a message by voting that tag (i.e. through a “+1” button next to the tag).

Then, messages could be filtered at any moment, by selecting one or more tags. Therefore, conversations could become organized while they are happening or afterwards.

Any comments welcome.
Best regards

**Comment by tox-user** [comment]
That is an interesting idea. Do you think people would be interested in adding those tags? There can be a lot of activity in some groups, so it might be hard for people to keep track of everything. What if people start assigning wrong tags?

**Comment by aramburu** [comment]
Good point, thanks. I am thinking of two type of tags, or two type of tag assignment. Let's say "message specific tags" and "topic tags". 
"Message specific tags" are assigned to a single message. For example, if someone announces a meeting. This could be tagged as "announcements", "important" or whatever.
On the other hand, "topic tags" are assigned or inherited automatically by means of simple heuristics. For example, if a message is sent very close in time to another, they are considered to belong to the same conversation, so the second message should inherit its topic-tags.  Then, if a group is discussing what movie they are going to see, the first message is tagged as "movie", and all messages that follow are tagged automatically. This automatic inheritance could be cleared after a timeout.

Wrong assignments could be address with the rating system. A tag could be voted positively or negatively by any group member. Tags could disappear after their rating falls below a given negative threshold.
